src/Lean/Server/FileWorker.lean

```diff
@@ -374,6 +381,9 @@ section Initialization
           -- note that because of `server.reportDelayMs`, we cannot simply set `maxDocVersion` here
           -- as that would allow outdated messages to be reported until the delay is over
         o.writeLspMessage msg |>.catchExceptions (fun _ => pure ())
+        if let .notification "$/lean/fileProgress" (some params) := msg then
+          if let some (params : LeanFileProgressParams) := fromJson? (toJson params) |>.toOption then
+            chanIsProcessing.send (! params.processing.isEmpty)
```

Comment:
Could be done in `reportSnapshots` without parsing? I do hope no-one else is sending out file progress notifs

Reply:
What if the version for the `fileProgress` notification is out-of-date?

Reply:
That shouldn't be a big concern as cancellation should prevent the reporter from doing that for any relevant amount of time but it's fine like this then